xmodule/capa_block.py

```diff
@@ -1753,7 +1819,13 @@ def submit_problem(self, data, override_time=False):
             self.attempts = self.attempts + 1
             self.lcp.done = True
             self.set_state_from_lcp()
-            self.set_score(self.score_from_lcp(self.lcp))
+
+            current_score = self.score_from_lcp(self.lcp)
+            self.score_history.append(current_score)
+            if self.enable_grading_method:
+                self.set_score_with_grading_method(current_score)
+            else:
+                self.set_score(current_score)
```

Comment:
Here you could not make such a change, since the `set_score` method is used in different parts of the code, which would affect the behavior. Instead I modified the new `set_score_with_grading_method` method to remove the else, like this:


```python
...

if self.enable_grading_method:
    current_score = self.set_score_with_grading_method(current_score)
self.set_score(current_score)

...
```